### PR TITLE
Keystone ldap integration

### PR DESCRIPTION
**Pull request by paulczar**

### CODE REVIEW

File: doc/ldap-demo.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# LDAP Demo
+
+Run a local ldap server, use docker for speed
```
Comment (dlundquist) [inline doc/ldap-demo.md]: {{Citation needed}}

File: doc/ldap-demo.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+install and setup ldap server:
+
+```
+$ apt-get install slapd ldap-utils
+$ dpkg-reconfigure slapd
+```
+
+* Omit OpenLDAP server configuration? No
+* DNS domain name : openstack.org
```
Comment (dlundquist) [inline doc/ldap-demo.md]: We shouldn't use someone else's domain for this, perhaps example.org?

File: doc/ldap-demo.md (hunk 1 above)
Comment (paulczar) [inline doc/ldap-demo.md]: heh, my initial testing was with a docker ldap server, forgot to clear this out of the instructions when I switched to installing one via apt packages.

**Comment by dlundquist** [comment]
Aside from using the OpenStack foundation's domain in our tests, my only concern duplicating the keystone config into two files. Could we use an additional file and config_dir instead?

File: roles/keystone/templates/etc/keystone/keystone.ldap.conf [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+#valid_days = 3650
+#ca_password = None
+
+[paste_deploy]
+config_file = /etc/keystone/keystone-paste.ini
+
+[revoke]
+driver = keystone.contrib.revoke.backends.sql.Revoke
+
+[ldap]
```
Comment (j2sol) [inline roles/keystone/templates/etc/keystone/keystone.ldap.conf]: Couldn't we just put this in an if block inside the existing template file?

Comment (j2sol) [inline roles/keystone/templates/etc/keystone/keystone.ldap.conf]: Oh, you did that in #957 ... why there as a separate PR?

**Comment by j2sol** [comment]
Looks okay to me, just odd that the one part I have an objection to is done in a different PR.

**Comment by j2sol** [comment]
This now looks good to me. :+1:

**Comment by paulczar** [comment]
this may not have actually passed tests ... got run by the new jenkins server I'm to test via PR.

**Comment by j2sol** [comment]
This is mostly obviated by the LDAP support we've added. We don't have an LDAP setup in vagrant at the moment, feel free to narrow the scope and re-open.

**Pull request closed without merging**